mantidproject/mantid
# Abins GAUSSIAN parser: use Input orientation if Standard unavailable

**Description of work**
Accept structure in "Input Orientation" from GAUSSIAN files if no "Standard Orientation" is available.

**Purpose of work**
Abins needs to read output files from a variety of electronic structure software. Most of these have badly specified and inconsistently formatting, so writing robust parsers is a rather iterative process as new problems are encountered.

Since https://github.com/mantidproject/mantid/pull/35821 a user has presented GAUSSIAN output with "Input orientation" blocks and no "Standard orientation" blocks; these would fail following the changes in that (merged) PR. (Which fixes the reading of files that have "Standard" but not "Input" orientation...)

**Report to:** Sanghamitra Mukhopadhyay

**Summary of work**
As "Standard orientation" comes after "Input orientation" in files that contain both blocks, a find_last() search that matches both lines should still get the (preferred) Standard block when it is present.

Fortunately in the previous work this search was refactored to use regex, so this is quite a simple fix.

**To test:**
Unit testing should be sufficient.

A new unit test is added to confirm that files with only "Input orientation" will work.
Existing tests should pass unchanged to confirm that "Standard orientation" is still preferred.

*There is no associated issue.*

The release note for #35821 may require tweaking

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
